lib/datasource/adoptium-java/index.spec.ts

```diff
@@ -10,7 +10,11 @@ const jre = loadFixture('jre.json');
 function getPath(page: number, imageType = 'jdk'): string {
   return `/v3/info/release_versions?page_size=${pageSize}&image_type=${imageType}&project=jdk&release_type=ga&sort_method=DATE&sort_order=DESC&vendor=adoptium&page=${page}`;
 }
-function* range(start: number, end: number): Generator<number, number, number> {
+
+function* range(
+  start: number,
+  end: number
+): Generator<number, number, number> | undefined {
```

Comment:
```suggestion
): IterableIterator<number> {
```
Maybe this works?

https://stackoverflow.com/a/68945873